server/game/core/card/EventCard.ts

```diff
@@ -63,4 +77,25 @@ export class EventCard extends EventCardParent {
         properties.cardName = this.title;
         this._eventAbility = new EventAbility(this.game, this, properties);
     }
+
+    /** Add a constant ability on the event card that decreases its cost under the given condition */
+    protected addDecreaseCostAbility(properties: IDecreaseEventCostAbilityProps<this>): void {
```

Comment:
Jabba's Rancor has the same self cost reduction type effect as Force Choke. Should this function be on a more generic card class?

Reply:
I wrote it this way because I didn't want to give event cards access to generic constant abilities, since this is the only type of constant ability an event card can have (currently). But it might make sense to make it a more generic helper that's available to more cards.